plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryTypeMapping.java

```diff
@@ -526,44 +533,128 @@ public void testTime()
                 .execute(getQueryRunner(), bigqueryViewCreateAndInsert("test.time"));
     }
 
-    @Test
-    public void testTimestampWithTimeZone()
+    @Test(dataProvider = "timestampsWithZoneDataProvider")
+    public void testTimestampWithTimeZone(ZoneId zoneId)
     {
-        SqlDataTypeTest.create()
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(zoneId.getId()))
+                .build();
+
+        testTimestampWithTimeZone("TIMESTAMP(6) WITH TIME ZONE")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test.timestamp_tz"))
+                .execute(getQueryRunner(), trinoCreateAsSelect(session, "test.timestamp_tz"))
+                .execute(getQueryRunner(), session, trinoCreateAsSelect("test.timestamp_tz"));
+
+        testTimestampWithTimeZone("TIMESTAMP")
+                .execute(getQueryRunner(), bigqueryCreateAndInsert("test.timestamp_tz"));
+    }
+
+    private SqlDataTypeTest testTimestampWithTimeZone(String inputType)
+    {
+        return SqlDataTypeTest.create()
                 // min value in BigQuery
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '0001-01-01 00:00:00.000000 UTC'",
+                .addRoundTrip(inputType, "TIMESTAMP '0001-01-01 00:00:00.000 UTC'",
```

Comment:
It would be nice to order by old -> new date so that we can understand missing dates easily.